Tests/SwiftParserTest/ExpressionTests.swift

```diff
@@ -2689,4 +2689,74 @@ final class StatementExpressionTests: XCTestCase {
 
     assertParse("_ = { (@Wrapper x) in }")
   }
+
+  func testClosureWithMissingParentheses() {
+    assertParse(
+      """
+      _ = { 1️⃣a: Int, b: Int 2️⃣in
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "expected '(' to start parameter clause",
+          fixIts: ["insert '('"]
+        ),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "expected ')' to end parameter clause",
+          fixIts: ["insert ')'"]
+        ),
+      ],
+      fixedSource: """
+        _ = { (a: Int, b: Int) in
+        }
+        """
+    )
+  }
+
+  func testClosureWithReturnArrowAndMissingParentheses() {
+    assertParse(
+      """
+      _ = { 1️⃣a: Int, b: Int 2️⃣ -> Int in
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "expected '(' to start parameter clause",
+          fixIts: ["insert '('"]
+        ),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "expected ')' to end parameter clause",
+          fixIts: ["insert ')'"]
+        ),
+      ],
+      fixedSource: """
+        _ = { (a: Int, b: Int) in
+        }
```

Comment:
I just checked and it looks like this test is failing. The fixed source should be 

```swift
        _ = { (a: Int, b: Int) -> Int in
        }
```

Reply:
Was looking into it.
Haven't found a fix yet 😄

Reply:
While debugging it hit me that `_ = { (a: Int, b: (Int)  -> Int) in }` is a valid actually valid swift code. 

So what do you think?
Should we let it be as is?

Otherwise we should stop parsing the type until we hit the arrow?

Reply:
Oh, I forgot that. I don’t think there’s an easy way to tell closure parameter parsing to stop at an arrow to diagnose this better. So, it’s fine for now, I would say.

Reply:
Added the test above to ensure that is correct.